operator/v1/types_dns.go

```diff
@@ -287,6 +287,23 @@ type ForwardPlugin struct {
 	//
 	// +optional
 	TransportConfig DNSTransportConfig `json:"transportConfig,omitempty"`
+
+
+	// protocolStrategy specifies the protocol to use for upstream DNS
+	// requests. Valid values for protocolStrategy are "TCP" or empty.
+	// If empty, the default behavior of CoreDNS is used, which is currently
+	// the following: use the protocol of the original client request, but
+	// this is subject to change over time. "TCP" specifies that the platform
+	// should use TCP for all upstream DNS requests, even if the client request
+	// uses UDP. "TCP" is useful for UDP-specific issues such as those created by
+	// non-compliant upstream resolvers, but may consume more bandwidth or
+	// increase DNS response time. Note that protocolStrategy only affects
+	// the protocol of DNS requests that CoreDNS makes to upstream resolvers.
+	// It does not affect the protocol of DNS requests between clients and
+	// CoreDNS.
+	//
+	// +optional
+	ProtocolStrategy ProtocolStrategy `json:"protocolStrategy,omitempty"`
```

Comment:
If this is a configuration API, please remove `omitempty` for discoverability

Reply:
I believe it's a configuration API:

A configuration API is one that is:
- typically cluster-scoped
  - True
- a singleton within the cluster
  - True
- managed by a cluster administrator only
  - True

But, the confusing thing is that this suggested configuration isn't the same style as all of the configuration around it, most of them have `omitempty`. Either someone classified these a Workload APIs, it wasn't done correctly, or this is new guidance. Are these inconsistencies a problem?

Reply:
Removed `omitempty` in https://github.com/openshift/api/compare/62d426d41b0310547f8fd8d7ade341fa2e3252f8..3c387292005b5d2e0b492e42a86d67370dfbb708

Reply:
> But, the confusing thing is that this suggested configuration isn't the same style as all of the configuration around it, most of them have omitempty. Either someone classified these a Workload APIs, it wasn't done correctly, or this is new guidance. Are these inconsistencies a problem?

Likely it wasn't done correctly in the past, or, this is new guidance since the time this was added. Incosistencies are not an issue, for new API fields, we must not repeat sins of the past